### PR TITLE
Removed the limitation on plan cache for partitioned tables

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs/pull/16359
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -1975,7 +1975,7 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
 
 从示例二结果可知，开启 `dynamic` 模式后，带 IndexJoin 的计划在执行查询时被选上。
 
-目前，静态和动态裁剪模式都不支持执行计划缓存。
+目前，静态裁剪模式既不支持 Prepare 语句执行计划缓存，也不支持非 Prepare 语句执行计划缓存。
```
Comment (songrijie) [inline partitioned-table.md:1978]:
```suggestion
目前，静态裁剪模式不支持执行计划缓存，无论是 Prepare 语句还是非 Prepare 语句。
```

**Review by songrijie — APPROVED** [review]
Rest LGTM

File: partitioned-table.md [hunk 2]
```diff
@@ -1975,7 +1975,7 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
 
 从示例二结果可知，开启 `dynamic` 模式后，带 IndexJoin 的计划在执行查询时被选上。
 
-目前，静态和动态裁剪模式都不支持执行计划缓存。
+目前，静态裁剪模式不支持执行计划缓存，无论是 Prepare 语句还是非 Prepare 语句。
```
Comment (Oreoxmt) [inline partitioned-table.md:1978]:
```suggestion
目前，静态裁剪模式不支持执行计划缓存，包括 Prepare 语句和非 Prepare 语句。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: optimizer-fix-controls.md [hunk 3]
```diff
@@ -26,6 +26,12 @@ SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
 
 ## Optimizer Fix Controls 参考
 
+### [`33031`](https://github.com/pingcap/tidb/issues/33031) <span class="version-mark">从 v8.0.0 版本开始引入</span>
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 此变量控制是否允许对分区表进行计划缓存。如果设置为 `ON`，则 [Prepared 语句执行计划缓存](/sql-prepared-plan-cache.md)和[非 Prepared 语句执行计划缓存](/sql-non-prepared-plan-cache.md)都不会对[分区表](/partitioned-table.md)启用。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:33]:
```suggestion
- 是否允许对分区表进行计划缓存。如果设置为 `ON`，则 [Prepared 语句执行计划缓存](/sql-prepared-plan-cache.md)和[非 Prepared 语句执行计划缓存](/sql-non-prepared-plan-cache.md)都不会对[分区表](/partitioned-table.md)启用。
```

File: optimizer-fix-controls.md [hunk 4]
```diff
@@ -26,6 +26,12 @@ SET SESSION tidb_opt_fix_control = '44262:ON,44389:ON';
 
 ## Optimizer Fix Controls 参考
 
+### [`33031`](https://github.com/pingcap/tidb/issues/33031) <span class="version-mark">从 v8.0.0 版本开始引入</span>
+
+- 默认值：`OFF`
+- 可选值：`ON`、`OFF`
+- 是否允许对分区表进行计划缓存。如果设置为 `ON`，则 [Prepared 语句执行计划缓存](/sql-prepared-plan-cache.md)和[非 Prepared 语句执行计划缓存](/sql-non-prepared-plan-cache.md)都不会对[分区表](/partitioned-table.md)启用。
```
Comment (hfxsd) [inline optimizer-fix-controls.md:33]:
```suggestion
- 是否允许对分区表进行计划缓存。如果设置为 `ON`，则 [Prepared 语句计划缓存](/sql-prepared-plan-cache.md)和[非 Prepared 语句计划缓存](/sql-non-prepared-plan-cache.md)都不会对[分区表](/partitioned-table.md)启用。
```

**Pull request merged**